### ISSUE TITLE
[1.9] Issue Burndown

### ISSUE THREAD

**Issue by enisoc** [kubernetes/sig-release#38]
This is a tracker for status updates on 1.9 issue burndown.

**Comment by jberkus** [comment]
Summary for Nov. 27

A slew of issues has been opened around failing tests, including:

New Test Failures (mostly e2e tests):
#56416
#56429
#56426
#56422
#56414

Old Test Failures in Progress:
#56262
#56235
#53020
#56091

Old Test Failures Not Getting Attention:
#56244
#56155
#56052

New Bugs:
#56357
#56399
#56357
#56348 (not clear that this should be in 1.9)

Old Bugs in Progress:
#56385
#56355

Old Bugs Not Getting Attention:

Non-release-blocker:
#47820

Special Cases: 

#54551: Advanced Auditing (https://github.com/kubernetes/kubernetes/issues/54551) really appears to be Not Ready for 1.9, depending on a bunch of PRs which either don't exist or aren't close to approval.  Some of those PRs might be separable from a 1.9 feature, but I'm not clear on which those are.  I've escalated to the feature/issue owner and one of the SIGs.  This includes some related issues:
  #53020
  #53006

#55978: this issue looks like it should be closed, given the code merges.  However, I haven't been able to get sig/node or JingXu to confirm that. 

#49480: CNI 0.6.0 this issue appears to have a huge chicken-egg problem with dependencies.  It's making progress, but slowly.

**Comment by jberkus** [comment]
Today's status: 

Summary for Nov. 28

20 total issues, so making progress

All "not getting attention" issues have been raised with the respective SIGs on Slack.

New Test Failures (<1 day old)
#54574 (old bug, just reopened)

Test Failures in Progress:
#56429
#56416
#56414
#56262
#56052
#55194
#47820

Test Failures Not Getting Attention:
#56244 (6 days)
#56426 (1 day)

New Bugs (<1 day old):

Bugs in Progress:
#56399
#56357
#56348

Bugs Not Getting Attention:
#56155 (3 days)
#55892 (5 days) (gave 24-hour deadline)

Special Cases:

#54551: Advanced Auditing (kubernetes/kubernetes#54551) Exception raised on burndown list. This includes some related issues:
#53020
#53006

#55978: has been closed.

#49480: CNI 0.6.0 Now waiting on figuring out packaging/deployment. Receiving attention.

#56504: kubeadm tracker.  I have not get gone through the kubeadm issues for status.  More later.

**Comment by jberkus** [comment]
So, based on today's meeting, let me give a breakdown of which issues appear to be blockers to me.  Please note that, for many of the ones not receiving enough attention, I also don't have feedback from the SIG on how critical they are, so I'm guessing based on the description & reports back.  Issues that I don't mention below are non-concerning.

## Bugs & Test Failures

### Red Tests/Bugs

Upgrade & Downgrade: Several failures, mostly on GKE.  Downgrade appears generally broken.
* 56244
* 56426 & 56429 & 56422 on GKE

### Yellow: 

Misc Bugs and Performance issues with proposed fixes:
* #56357, polling problem, already has proposed fix
* #55892, detach broken in AWS, already has proposed PR which needs adding

Tests that need test fixes:
* #56416, network partition test, claim is that there's a defect in the test.
* #56414

Performance Test issues, not possible to tell if these are actual performance issues
or issues with the test platform:
* #56052

## Features

### Blocking

Kubeadm just added a whole list of milestone items (#56504).  However, it is opaque to me which of those is actually a blocker, and which is in progress.

CNI 0.6.0 (#49480): the code has been changed, but how this works for deployments and packaging has not been resolved.

### Non-Blocking

AWS NVME Support: #56155, in progress, but if we don't have it, it's not a feature that worked before.  See Exception request.

Advanced Audit Logging: includes #53020, #53006, #54551.  This whole feature can be pushed to 1.10 if it's not ready on time.  See Exception request.

**Comment by zacharysarah** [comment]
## Docs

Friday, December 1 is the [deadline for all docs PRs](https://github.com/kubernetes/features/blob/master/release-1.9/release-1.9.md#docs) to be reviewable.

According to the [feature tracking spreadsheet](https://docs.google.com/spreadsheets/d/1WmMJmqLvfIP8ERqgLtkKuE_Q2sVxX8ZrEcNxlVIJnNc/edit#gid=0), these issues still need to [open a docs PR](https://github.com/kubernetes/features/blob/master/release-1.9/release-1.9.md#docs) (due 11/22):

https://github.com/kubernetes/features/issues/365
https://github.com/kubernetes/features/issues/353
https://github.com/kubernetes/features/issues/507
https://github.com/kubernetes/features/issues/491
https://github.com/kubernetes/features/issues/178

/cc @enisoc @idvoretskyi

**Comment by jberkus** [comment]
Status update for Friday, Dec 1:

Status of PRs and Bugs

I'm adding PRs to this burndown, because I'm concerned about PRs which haven't yet been merged.

The good news is that we're down to 18 issues and 10 PRs.  The bad news is that two of those 18 issues are tracking issues.  This status is updated before the extensions are due.

## Red
*bugs/failures which appear user-visible and for which no approved PR exists*
*PRs which appear required and do not have all sign-offs or are stalled*

Issues:

* #56594 Kubeadm test (new failure)
* #56357 Performance Polling Issue (PR submitted, but see below)
* #56244 Downgrade still failing (just got owner)
* #56052 & 55194 Scale tests still failing.  (Nagged issue owner/SIG)

PRs:

* #56478 fix for polling issue, apparently has major bugs
  (not tagged with 1.9 milestone) (SIG polled)
* #56513 upgrade/downgrade fixes
* #56513 Kubeadm Upgrade/Downgrade bugs

## Yellow
*bugs/failures which either appear minor, or for which a PR in good shape exists*
*PRs which are approved but are waiting to be merged*

Issues:
* #56422 GKE Downgrade Failure (only affects downgrading 2 versions)
* #56429 GKE ResourceQuota still failing (similar fix as per other GKE fixes)
* #54574 Horiz Autoscaling Test Failure (PR submitted)
* #56532 Storage errors (new bug, PR submitted)

PRs:
* #56639 add test for volume resizing
* #56598 fix PV bug
* #56576 fix metadata agent
* #56533 fix new storage errors
* #50603 content negotiation bug (have queried owner because it doesn't appear 
  to have anything tied to 1.9, suggested removal)

## Green
*issues which appear resolved but are not yet closed/removed.*
*PRs which are going to be removed from milestone but have not been*

Issues:
* #56416
* #56414
* #47820
* #53006
  

## Exception Status & Special Cases

Advanced Auditing: due today.
Issues: #54551, #53020
PRs: #56638

CNI 0.6.0: (#49480): see above.  Also there seems to be no leadership on figuring this out.

AWS NVME Support: due today

Update Dashboard: due today, appears to be waiting only on owner approval.

Autoprobing Network-ID: due today, appears to just be waiting on owner approval.

**Comment by jberkus** [comment]
Status of PRs and Bugs

Down to 10 issues, and 14 PRs.  Many of those PRs are random changes which happen
to be approved now and owners are hoping to get them into 1.9; none of the new
PRs look like blockers.

Remaining critical issues appear to be the Azure breakage, failure to test downgrade,
and CNI 0.6.0.


## Red
*bugs/failures which appear user-visible and for which no approved PR exists*
*PRs which appear required and do not have all sign-offs or are stalled*

Issues:

* https://github.com/kubernetes/kubernetes/issues/56686 Azure loadbalancer
  (new bug, opened Dec 1, escalated to SIG for update)
* https://github.com/kubernetes/kubernetes/issues/56416 
  Network partition healing fail 
  (had PR, fix didn't work)
* https://github.com/kubernetes/kubernetes/issues/56244 
  Downgrade still failing
  (assigned to developer, but no updates, escalated)

PRs:

* https://github.com/kubernetes/kubernetes/pull/56709 
  fixes issue which breaks https://github.com/kubernetes/kubernetes/pull/55558
* https://github.com/kubernetes/kubernetes/pull/56688
  fixes Azure loadbalancer issue (above) but broken in beta 1
  (no ETA on fixed fix)

## Yellow
*bugs/failures which either appear minor, or for which a PR in good shape exists*
*PRs which are approved but are waiting to be merged*

Issues:
* https://github.com/kubernetes/kubernetes/issues/56426 
  GKE Downgrade Failure (possibly fixed)
* https://github.com/kubernetes/kubernetes/issues/56357
  Performance Polling Issue 
  (See issue.  ETA is 1.9.1, team is trying for 1.9.0.  SIG does not consider 
  this issue a showstopper)
* https://github.com/kubernetes/kubernetes/issues/56052
  https://github.com/kubernetes/kubernetes/issues/55194
  Scale tests still failing.  
  (According to SIG on 12/1, test issue not scale issue.  May
  be fixed, too many test flakes to tell yet)


PRs:
* #56478 fix for polling issue, apparently has major bugs
  (not tagged with 1.9 milestone) (See Issue for status notes)
* https://github.com/kubernetes/kubernetes/pull/50603 
  content negotiation bug (no response from owning
  sig on whether this should actually be in 1.9
  or ETA.  Suggest removal from milestone)
* https://github.com/kubernetes/kubernetes/pull/56802 
  Bump version 
* https://github.com/kubernetes/kubernetes/pull/56785
  Possible fix for scaling tests
* https://github.com/kubernetes/kubernetes/pull/56623
  IPvs fix (ready to merge)
* https://github.com/kubernetes/kubernetes/pull/56444
  Add months for cert expiration
  (ETA 12/4, currently not passing tests, but is
  new feature and could be dropped)
* https://github.com/kubernetes/kubernetes/pull/55360
  fix metadata iptables something?
  (appears ready, missing labels)


## Green
*issues which appear resolved but are not yet closed/removed.*
*PRs which are not approved for milestone
 or are waiting on RC to merge*

PRs:
* https://github.com/kubernetes/kubernetes/pull/56599
  Kubeadm version stamp
  (cannot be merged until RC)
* https://github.com/kubernetes/kubernetes/pull/56687
  Alpha/beta feature default for admissioncontroller
  (not currently approved for milestone)
* https://github.com/kubernetes/kubernetes/pull/56643
  Test fix for CoreOS metrics
  (not currently approved for milestone)
* https://github.com/kubernetes/kubernetes/pull/56519
  Make sure node is ready
  (ready, not approved for milestone)
* https://github.com/kubernetes/kubernetes/pull/55448
  Fix GKE system spec
  (ready, not approved for milestone)

## Exception Status & Special Cases

Advanced Auditing: due 12/1

* Issues: https://github.com/kubernetes/kubernetes/issues/55194 (tracking)
  https://github.com/kubernetes/kubernetes/issues/53020 (testing, no status update,
  queried issue owner)
* PRs: 

CNI 0.6.0: (https://github.com/kubernetes/kubernetes/issues/49480): issue still lacks
an "owner" who is going to make sure that the deploy & packaging changes happen.

Kubeadm Changes (https://github.com/kubernetes/kubernetes/issues/56504): according to SIG,
is almost complete with all changes in except for one open PR (https://github.com/kubernetes/kubernetes/pull/56599)

AWS NVME Support: Completed

Update Dashboard: Completed

Autoprobing Network-ID: Completes

**Comment by jberkus** [comment]
Status of PRs and Bugs

Currently 13 issues (including new bugs).  There are 16 PRs, but quite a number
of these are pending PRs which do not have SIG approval for the milestone.  I've 
put the unapproved PRs in "Green", because the idea is that most of them are
more likely to simply be dropped.

Remaining critical issues appear to be failure to test downgrade,
and CNI 0.6.0.


## Red
*bugs/failures which appear user-visible and for which no approved PR exists*
*PRs which appear required and do not have all sign-offs or are stalled*

Issues:

* https://github.com/kubernetes/kubernetes/issues/56813
  CSI teardown issue
  (new bug, has PR which has not been reviewed)
* https://github.com/kubernetes/kubernetes/issues/56833
  GCE cloud provider issue
  (has more than one competing PR, see below)
* https://github.com/kubernetes/kubernetes/issues/56416
  Network partition healing fail
  (had PR, fix didn't work)
* https://github.com/kubernetes/kubernetes/issues/56244
  Downgrade still failing
  (in progress, some test improvement, no ETA)

PRs:

* https://github.com/kubernetes/kubernetes/pull/56709
  fixes issue which breaks https://github.com/kubernetes/kubernetes/pull/55558
* https://github.com/kubernetes/kubernetes/pull/56836
  Fix for CSI teardown issue
  (needs review)
* https://github.com/kubernetes/kubernetes/pull/56837
  https://github.com/kubernetes/kubernetes/pull/56861
  Competing fixes for GCE issue.
  (needs review, decision on competing approaches)

## Yellow
*bugs/failures which either appear minor, or for which a PR in good shape exists*
*PRs which are approved but are waiting to be merged*

Issues:
* https://github.com/kubernetes/kubernetes/issues/56265
  Statefulset PVC issue
  (new bug, has approved PR)
* https://github.com/kubernetes/kubernetes/issues/56357
  Performance Polling Issue
  (See issue.  ETA is 1.9.1, team is trying for 1.9.0.  SIG does not consider
  this issue a showstopper)
* https://github.com/kubernetes/kubernetes/issues/56686
  Azure loadbalancer
* https://github.com/kubernetes/kubernetes/issues/47820
  Network Partition test failing
  (SIG says test should be non-blocking, but has not changed
   status of test or removed issue)

PRs:
* https://github.com/kubernetes/kubernetes/pull/56577
  Fix for Statefulset PVC issue (approved)
* #56478 fix for polling issue, apparently has major bugs
  (not tagged with 1.9 milestone) (See Issue for status notes)
* https://github.com/kubernetes/kubernetes/pull/56824
  fix for Yaml quoting issue
  (tiny, approved)
* https://github.com/kubernetes/kubernetes/pull/50603
  content negotiation bug (no response from owning
  sig on whether this should actually be in 1.9
  or ETA.  Suggest removal from milestone)
  
## Green
*issues which appear resolved but are not yet closed/removed.*
*PRs which are not approved for milestone
 or are waiting on RC to merge*

Issues:
* https://github.com/kubernetes/kubernetes/issues/56052
  https://github.com/kubernetes/kubernetes/issues/55194
  Scale tests.
  (One test has passed and the other is expected to pass)

PRs:
* https://github.com/kubernetes/kubernetes/pull/56599
  Kubeadm version stamp
  (cannot be merged until RC)
* https://github.com/kubernetes/kubernetes/pull/56687
  Alpha/beta feature default for admissioncontroller
  (not currently approved for milestone)
* https://github.com/kubernetes/kubernetes/pull/56643
  Test fix for CoreOS metrics
  (not approved, not approved for milestone)
* https://github.com/kubernetes/kubernetes/pull/56519
  Make sure node is ready
  (ready, not approved for milestone)
* https://github.com/kubernetes/kubernetes/pull/55448
  Fix GKE system spec
  (ready, not approved for milestone)
* https://github.com/kubernetes/kubernetes/pull/56538
  cleanup useless functions?
  (ready, not approved for milestone)
* https://github.com/kubernetes/kubernetes/pull/56390
  test proxy mode
  (ready, not approved for milestone)
* https://github.com/kubernetes/kubernetes/pull/56298
  Addressing comments from code review
  (ready, not approved for milestone)
* https://github.com/kubernetes/kubernetes/pull/55925
  Fix ServiceNodeExclusion version
  (ready, not approved for milestone)
* https://github.com/kubernetes/kubernetes/pull/55467
  Remove duplicate code
  (not really 1.9)
* https://github.com/kubernetes/kubernetes/pull/55806
  Remove unused function
  (ready, not approved, not labeled)

## Exception Status & Special Cases

Advanced Auditing: due 12/1
Issues: https://github.com/kubernetes/kubernetes/issues/55194 (tracking)
All PRs complete.  Author says one more PR is required to fix one GCE-specific issue.

CNI 0.6.0: (https://github.com/kubernetes/kubernetes/issues/49480): issue still lacks
an "owner" who is going to make sure that the deploy & packaging changes happen.

Kubeadm Changes (https://github.com/kubernetes/kubernetes/issues/56504): according to SIG,
is almost complete with all changes in except for one open PR
(https://github.com/kubernetes/kubernetes/pull/56599) which has to be finished
at release.

AWS NVME Support: Completed

Update Dashboard: Completed

Autoprobing Network-ID: Completed

**Comment by zacharysarah** [comment]
# Docs

Status of PRs, cribbing shamelessly from @jberkus 

## Red
_PRs which do not have a docs and/or tech review, or whose opener is unresponsive_

PR | Description
---|---
https://github.com/kubernetes/website/pull/6260 | No tech reviewer assigned
https://github.com/kubernetes/website/pull/6550 | No response to docs review, no tech reviewer assigned
https://github.com/kubernetes/website/pull/6536 | No response to docs review
https://github.com/kubernetes/website/pull/6550 | No response to docs review, no tech reviewer assigned
https://github.com/kubernetes/website/pull/6555 | No response to docs review, no tech reviewer assigned

## Yellow
_PRs which have docs/tech reviewers but require action prior to merging_

PR | Description
---|---
https://github.com/kubernetes/website/pull/6368 | PR merged prematurely, new PR required (doc/tech reviews received) (1)
https://github.com/kubernetes/website/pull/6371 | Respond to tech review
https://github.com/kubernetes/website/pull/6415 | Needs response from tech reviewers
https://github.com/kubernetes/website/pull/6427 | Failing CI/CD check
https://github.com/kubernetes/website/pull/6465 | Changes need to be made upstream for autogeneration, in kubernetes/kubernetes (2)
https://github.com/kubernetes/website/pull/6485 | Needs response from tech reviewers
https://github.com/kubernetes/website/pull/6498 | Changes need to be made upstream for autogeneration, in kubernetes/kubernetes (2)

1: I'm working with @lavalamp to complete review by Monday 12/11
2: I will open these PRs tomorrow, Thursday 11/7

## Green
_PRs that have been merged to the release megabranch, or otherwise require no pre-release action_

PR | Description
---|---
https://github.com/kubernetes/website/pull/6066 | Done
https://github.com/kubernetes/website/pull/6103 | Needs autogeneration (PR # pending)
https://github.com/kubernetes/website/pull/6392 | Done
https://github.com/kubernetes/website/pull/6463 | Done
https://github.com/kubernetes/website/pull/6474 | Done
https://github.com/kubernetes/website/pull/6479 | Done
https://github.com/kubernetes/website/pull/6487 | Done
https://github.com/kubernetes/website/pull/6494 | Done
https://github.com/kubernetes/website/pull/6496 | Done
https://github.com/kubernetes/website/pull/6504 | Done
https://github.com/kubernetes/website/pull/6518 | Done
https://github.com/kubernetes/website/pull/6519 | Merge after release
https://github.com/kubernetes/website/pull/6520 | Merge after release
https://github.com/kubernetes/website/pull/6521 | Merge after release
https://github.com/kubernetes/website/pull/6522 | Merge after release
https://github.com/kubernetes/website/pull/6553 | Done
https://github.com/kubernetes/website/pull/6554 | Done
https://github.com/kubernetes/website/pull/6558 | Done

**Comment by jberkus** [comment]
Status of PRs and Bugs

Currently 10 issues (including new bugs).  There are 11 PRs.

  I've 
put the unapproved PRs in "Green", because the idea is that most of them are
more likely to simply be dropped.

Remaining critical issues appear to be failure to test downgrade,
and CNI 0.6.0.

(and thanks Zachary for showing that I could be using markdown tables for this!)

## Red
*bugs/failures which appear user-visible and for which no approved PR exists*
*PRs which appear required and do not have all sign-offs or are stalled*

Issues:

| issue | summary | status |
|:----- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/issues/56244 |  Downgrade still failing | (debugging, no ETA) |
| https://github.com/kubernetes/kubernetes/issues/56879 | Parallel downgrade failing | (newly escalated) |
| https://github.com/kubernetes/kubernetes/issues/56426 | GKE downgrade failure | was resolved, now failing again, in progress |
| https://github.com/kubernetes/kubernetes/issues/56416 | Network partition healing fail | had PR, fix didn't work |


PRs:

| PR | summary | status |
|:------- |:------- |:------ |

## Yellow
*bugs/failures which either appear minor, or for which a PR in good shape exists*
*PRs which are approved but are waiting to be merged*

Issues:

| issue | summary | status |
|:----- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/issues/56357 | Performance Polling Issue | ETA 1.9.1, not showstopper |
| https://github.com/kubernetes/kubernetes/issues/47820 | Network Partition test fail | SIG says non-blocking but has not removed |
| https://github.com/kubernetes/kubernetes/issues/53207 | Kubelet status update hang | newly escalated, has PR |

PRs:

| PR | summary | status |
|:----- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/pull/56824 | fix for Yaml quoting issue | tiny, approved |
| https://github.com/kubernetes/kubernetes/pull/50603 | fix content negotiation bug | not ready, no ETA, non-blocker |
| https://github.com/kubernetes/kubernetes/pull/56967 | update cadvisor PR for issue #53207 | 2/3 approvals |
| https://github.com/kubernetes/kubernetes/pull/56970 | cherry-pick for autoscaler | 1/3 approvals |
| https://github.com/kubernetes/kubernetes/pull/56942 | doc updates for 1.9 | needs approval |
| https://github.com/kubernetes/kubernetes/pull/56679 | fix azure storage delay | approved |
  
## Green
*issues which appear resolved but are not yet closed/removed.*
*PRs which are not approved for milestone
 or are waiting on RC to merge*

| issue | summary | status |
|:----- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/issues/56813 | CSI teardown issue | PR merged, close? |

PRs:

| PR | summary | status |
|:----- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/pull/56959 | fix lifecycle messaging | not approved for milestone, appears minor |
| https://github.com/kubernetes/kubernetes/pull/56918 | cherry-pick to fix Azure LB issue | ready, not approved for milestone |
| https://github.com/kubernetes/kubernetes/pull/56599 | Kubeadm version stamp | merge @ RC |
| https://github.com/kubernetes/kubernetes/pull/56390 | text IPVS proxy | ready, not approved for milestone |
| https://github.com/kubernetes/kubernetes/pull/55925 | Fix ServiceNodeExclusion version | ready, not approved for milestone |

## Exception Status & Special Cases

Advanced Auditing:  Complete 

CNI 0.6.0: (https://github.com/kubernetes/kubernetes/issues/49480): issue still lacks
an "owner" who is going to make sure that the deploy & packaging changes happen.

Kubeadm Changes (https://github.com/kubernetes/kubernetes/issues/56504): according to SIG,
is almost complete with all changes in except for one open PR
(https://github.com/kubernetes/kubernetes/pull/56599) which has to be finished
at release.

AWS NVME Support: Completed

Update Dashboard: Completed

Autoprobing Network-ID: Completed

**Comment by jberkus** [comment]
Status of PRs and Bugs

Currently 9 issues (including new bugs).  There are 15 PRs, including several partial fixes
for outstanding issues.  I've 
put the unapproved PRs in "Green", because the idea is that most of them are
more likely to simply be dropped.

Remaining critical issues appear to be failure to test downgrade,
and CNI 0.6.0.

(and thanks Zachary for showing that I could be using markdown tables for this!)

## Red
*bugs/failures which appear user-visible and for which no approved PR exists*
*PRs which appear required and do not have all sign-offs or are stalled*

Issues:

| issue | summary | status |
|:----- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/issues/56244 |  Downgrade still failing | (debugging, no ETA) |
| https://github.com/kubernetes/kubernetes/issues/56879 | Parallel downgrade failing | (newly escalated) |
| https://github.com/kubernetes/kubernetes/issues/56426 | GKE downgrade failure | was resolved, now failing again, in progress |
| https://github.com/kubernetes/kubernetes/issues/56416 | Network partition healing fail | had PR, fix didn't work |


PRs:

| PR | summary | status |
|:------- |:------- |:------ |

## Yellow
*bugs/failures which either appear minor, or for which a PR in good shape exists*
*PRs which are approved but are waiting to be merged*

Issues:

| issue | summary | status |
|:----- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/issues/56357 | Performance Polling Issue | ETA 1.9.1, not showstopper |
| https://github.com/kubernetes/kubernetes/issues/53207 | Kubelet status update hang | newly escalated, has PR |

PRs:

| PR | summary | status |
|:----- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/pull/56824 | fix for Yaml quoting issue | tiny, approved |
| https://github.com/kubernetes/kubernetes/pull/50603 | fix content negotiation bug | not ready, no ETA, non-blocker |
| https://github.com/kubernetes/kubernetes/pull/56967 | update cadvisor PR for issue #53207 | 2/3 approvals |
| https://github.com/kubernetes/kubernetes/pull/56970 | cherry-pick for autoscaler | 1/3 approvals |
| https://github.com/kubernetes/kubernetes/pull/56942 | doc updates for 1.9 | needs approval |
| https://github.com/kubernetes/kubernetes/pull/56679 | fix azure storage delay | approved |
| https://github.com/kubernetes/kubernetes/pull/57046 | heapster version bump | new, 2/3 approvals |
| https://github.com/kubernetes/kubernetes/pull/57043 | kubeadm etcd downgrade fix | new, approved but needs retest |
| https://github.com/kubernetes/kubernetes/pull/57023 | kubelet CRI defined in cadvisor | new, not approved for milestone, but author says it's a must-fix |
  
## Green
*issues which appear resolved but are not yet closed/removed.*
*PRs which are not approved for milestone
 or are waiting on RC to merge*

| issue | summary | status |
|:----- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/issues/47820 | Network Partition test fail | SIG claims test removed |

PRs:

| PR | summary | status |
|:----- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/pull/56959 | fix lifecycle messaging | not approved for milestone, appears minor |
| https://github.com/kubernetes/kubernetes/pull/56918 | cherry-pick to fix Azure LB issue | ready, not approved for milestone |
| https://github.com/kubernetes/kubernetes/pull/56599 | Kubeadm version stamp | merge @ RC |
| https://github.com/kubernetes/kubernetes/pull/56390 | text IPVS proxy | ready, not approved for milestone |
| https://github.com/kubernetes/kubernetes/pull/55925 | Fix ServiceNodeExclusion version | ready, not approved for milestone |
| https://github.com/kubernetes/kubernetes/pull/56858 | e2e test for custom metrics | ready, not approved for milestone |

## Exception Status & Special Cases

Advanced Auditing:  Complete 

CNI 0.6.0: (https://github.com/kubernetes/kubernetes/issues/49480): issue still lacks
an "owner" who is going to make sure that the deploy & packaging changes happen.  There have been
some updates on this, but a critical outstanding issue is making sure that people can still install version 1.8 with the correct CNI.

Kubeadm Changes (https://github.com/kubernetes/kubernetes/issues/56504): according to SIG,
is almost complete with all changes in except for one open PR
(https://github.com/kubernetes/kubernetes/pull/56599) which has to be finished
at release.

AWS NVME Support: Completed

Update Dashboard: Completed

Autoprobing Network-ID: Completed

**Comment by jberkus** [comment]
R status Update

This is the Sudden Death Overtime tracking of PRs

## Red

* critical PRs fixing major breakage issues with 1.9.  Appropriate SIGs have been queried for status today.*

| PR | summary | status |
|:------- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/pull/57043 | kubeadm etcd downgrade fix | new, approved but needs retest |
| https://github.com/kubernetes/kubernetes/pull/57023 | kubelet CRI defined in cadvisor | new, not approved for milestone, but author says it's a must-fix |
| https://github.com/kubernetes/kubernetes/pull/56967 | update cadvisor PR for issue #53207 | 2/3 approvals |
| https://github.com/kubernetes/kubernetes/pull/57052 | fix GCP permissions on kube-env | new, fixes fix that didn't fix, needs milestone approval |


## Yellow

* PRs with status updates, which are non-blockers but have not been kicked out of 1.9 *

| PR | summary | status |
|:------- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/pull/56970 | cherry-pick for autoscaler | should merge automatically |
| https://github.com/kubernetes/kubernetes/pull/56959 | fix lifecycle messaging | all approved but SIG owner used wrong labels, queries |
| https://github.com/kubernetes/kubernetes/pull/50603 | fix content negotiation bug | approved, waiting for merge |
| https://github.com/kubernetes/kubernetes/pull/56824 | fix for Yaml quoting issue | tiny, approved |
| https://github.com/kubernetes/kubernetes/pull/56679 | fix azure storage delay | approved, waiting for merge |



## Green

* PRs which appear non-critical, have been bumped by downgrading them to priority/important-soon
or can't be bumped because they're not approved*

| PR | summary | status |
|:------- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/pull/55925 | Fix ServiceNodeExclusion version | downgraded, but bot not kicking it out? |
| https://github.com/kubernetes/kubernetes/pull/56390 | text IPVS proxy | ready, not approved for milestone |
| https://github.com/kubernetes/kubernetes/pull/56918 | azure nodeport probe | blocked by member of SIG |

**Comment by jberkus** [comment]
Ok, I should have created hard deadlines much earlier.

We're down to two PRs now:

## Red

| PR | summary | status |
|:------- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/pull/57023 | kubelet CRI defined in cadvisor | new, not approved for milestone, but author says it's a must-fix |

## Yellow

None

## Green


| PR | summary | status |
|:------- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/pull/56858 | Custom Metrics Test | not approved, not critical |

**Comment by jberkus** [comment]
Status of PRs and Bugs

Final burndown

Remaining critical issues appear to be failure to test downgrade,
and CNI 0.6.0.

## Red
*bugs/failures which appear critical and for which no approved PR exists*
*PRs which appear required and do not have all sign-offs or are stalled*

Issues:

| issue | summary | status |
|:----- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/issues/56244 |  Downgrade still failing | (debugging, no ETA) |
| https://github.com/kubernetes/kubernetes/issues/56879 | Parallel downgrade failing | (newly escalated) |
| https://github.com/kubernetes/kubernetes/issues/57047 | Stackdriver logging issue | severity unclear, no PR, related to downgrade fail |
| https://github.com/kubernetes/kubernetes/issues/57005 | crio.sock mismatch | severity unclear, has unapproved PR |

PRs:

| PR | summary | status |
|:------- |:------- |:------ |
|https://github.com/kubernetes/kubernetes/pull/57023| fix for 57005 | not ready, not approved |

## Yellow
*bugs/failures which either appear to be not showstoppers, or for which a PR in good shape exists*
*PRs which are approved but are waiting to be merged*

Issues:

| issue | summary | status |
|:----- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/issues/56261 | Node deletion error | severity unclear, has unapproved PR |

PRs:

| PR | summary | status |
|:----- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/pull/56622 | Fix for 56261 | approved, waiting to merge |
  
## Green
*non-blocker issues*
*PRs which are not approved for milestone
 or are waiting on RC to merge*

| issue | summary | status |
|:----- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/issues/56052 | scale tests failing again | issue in milestone for tracking, non-blocker |

PRs:

| PR | summary | status |
|:----- |:------- |:------ |

## Exception Status & Special Cases

Advanced Auditing:  Complete 

CNI 0.6.0: (https://github.com/kubernetes/kubernetes/issues/49480): issue still lacks
an "owner" who is going to make sure that the deploy & packaging changes happen.  There have been
some updates on this, but a critical outstanding issue is making sure that people can still install version 1.8 with the correct CNI.

Kubeadm Changes (https://github.com/kubernetes/kubernetes/issues/56504): according to SIG,
is almost complete with all changes in except for one open PR
(https://github.com/kubernetes/kubernetes/pull/56599) which has to be finished
at release.

AWS NVME Support: Completed

Update Dashboard: Completed

Autoprobing Network-ID: Completed

**Comment by jberkus** [comment]
Status of PRs and Bugs

Final Final Final burndown

Yay downgrade is fixed, now it's really just CNI

Giving up on usual format, because there's really just a handful of things to fix.

## Stackdriver issue: has two PRs, both approved

| PR/issue | summary | status |
|:------- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/issues/57047 | logging test failure | approved, has PR |
| https://github.com/kubernetes/kubernetes/pull/57174 | cherrypick to fix issue | approved |
| https://github.com/kubernetes/kubernetes/pull/57172 | PR to fix issue | approved |

## CNI 0.6.0

| PR/issue | summary | status |
|:------- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/issues/49480 | update to CNI 0.6.0 | almost complete |
| https://github.com/kubernetes/release/pull/486 | fix rpm build for release |

## Scalability test flakiness

| PR/issue | summary | status |
|:------- |:------- |:------ |
| https://github.com/kubernetes/kubernetes/issues/56052 | scalability test failures | SIG treating this as test flake rather than real issue |
